src/table/utils/__tests__/find-visible-rows.spec.ts

```diff
@@ -1,196 +1,325 @@
+        getBoundingClientRect: (): Rect => bodyRect,
+        querySelectorAll: () => cells,
+      };
+      rootElement = {
+        querySelector: () => {
+          return tableBody;
+        },
+      } as unknown as HTMLDivElement;
+    };
+
     beforeEach(() => {
-      totalsRowRec = { y: 50, height: 20 };
-      totalsRow = { getBoundingClientRect: (): Rect => totalsRowRec };
-      totalsPosition = { atTop: true, atBottom: false };
+      bodyRect = { y: 0, height: 495 };
     });
 
-    it('should return correct visible row indices when the whole first row is visible', () => {
-      createElements(70);
-      const { visibleRowStartIndex, visibleRowEndIndex } = findVisibleRows(rootElement, totalsPosition);
-      expect(visibleRowStartIndex).toEqual(0);
-      expect(visibleRowEndIndex).toEqual(6);
-    });
+    describe('scrollTopRatio = 1', () => {
+      beforeEach(() => {
+        tableBody = {};
+        viewService = {
+          scrollTopRatio: 1,
+          visibleTop: 0,
+          visibleHeight: 11,
+          page: 0,
+          rowsPerPage: 100,
+          qTop: 0,
+          qHeight: 100,
+        };
+      });
 
-    it('should return correct visible row indices when 50% of the first row is visible', () => {
-      createElements(60);
-      const { visibleRowStartIndex, visibleRowEndIndex } = findVisibleRows(rootElement, totalsPosition);
-      expect(visibleRowStartIndex).toEqual(0);
-      expect(visibleRowEndIndex).toEqual(6);
-    });
+      it('should return correct visible row indices when the whole first row is visible', () => {
+        createElements(0);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findVirtualizedVisibleRows(rootElement, viewService);
+        expect(visibleRowStartIndex).toEqual(0);
+        expect(visibleRowEndIndex).toEqual(10);
+      });
 
-    it('should return correct visible row indices when less than 50% of the first row is half visible', () => {
-      createElements(59);
-      const { visibleRowStartIndex, visibleRowEndIndex } = findVisibleRows(rootElement, totalsPosition);
-      expect(visibleRowStartIndex).toEqual(1);
-      expect(visibleRowEndIndex).toEqual(6);
-    });
+      it('should return correct visible row indices when most of the first row is visible', () => {
+        createElements(-4);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findVirtualizedVisibleRows(rootElement, viewService);
+        expect(visibleRowStartIndex).toEqual(0);
+        expect(visibleRowEndIndex).toEqual(10);
+      });
 
-    it('should return correct visible row indices when the whole last row is visible', () => {
-      createElements(-125);
-      const { visibleRowStartIndex, visibleRowEndIndex } = findVisibleRows(rootElement, totalsPosition);
-      expect(visibleRowStartIndex).toEqual(6);
-      expect(visibleRowEndIndex).toEqual(9);
+      it('should return correct visible row indices when less than 80% of the first row is visible', () => {
```

Comment:
is it less than 80%? I thought it was 50%

Reply:
Actually it is not 80%, it is 4px (which is padding) of the cell outside of the viewport. I will update the text.